browse/routes/src.py

```diff
@@ -26,10 +28,18 @@
     data['abs_meta'] = docmeta
     data['arxiv_id'] = docmeta.arxiv_identifier
     data['anc_file_list'] = get_article_store().get_ancillary_files(docmeta)
+
+    headers: Dict[str,str]={}
+    headers.update(add_surrogate_key(headers,["anc",f"paper-id-{docmeta.arxiv_identifier.id}"]))
+    if re.search(r'\d+v\d+', arxiv_id): #get abs always adds a verion onto the id
```

Comment:
If you are checking the shape of arxiv id, rather than using regex, feed it to arXiv Identifier, catch parsing error and the id has the version. As is now, this does not support the old-style arXiv ID. This also future proofs your code if arXiv ever changes the ID format. (tl;dr DRY)

`
from arxiv.identifier import Identifier as arXivID, IdentifierException
def make_identifier(arxiv_id: str) -> arXivID | None:
    """Make an arXiv ID. If it is invalid, return None."""
    try:
        return arXivID(arxiv_id)
    except IdentifierException:
        pass
    return None

    xid = make_identifier(arxiv_id)
    if xid and xid.has_version():
        ...
`

Reply:
good idea, thanks!